pyanaconda/modules/services/installation.py

```diff
@@ -48,23 +50,32 @@ def __init__(self, sysroot, setup_on_boot):
     def name(self):
         return "Configure Initial Setup"
 
-    def run(self):
-        unit_name = "initial-setup.service"
-        if self._setup_on_boot == SetupOnBootAction.DISABLED:
-            log.debug("Initial Setup will be disabled.")
-            util.disable_service(unit_name)
-            return
-
-        if not os.path.exists(os.path.join(self._sysroot, "lib/systemd/system/", unit_name)):
+    def _enable_service(self):
+        """Enable the Initial Setup service."""
+        if util.unit_file_exists(self.INITIAL_SETUP_UNIT_NAME, root=self._sysroot):
+            util.enable_service(self.INITIAL_SETUP_UNIT_NAME, root=self._sysroot)
+        else:
             log.debug("Initial Setup will not be started on first boot, because "
-                      "it's unit file (%s) is not installed.", unit_name)
-            return
+                      "it's unit file (%s) is not installed.", self.INITIAL_SETUP_UNIT_NAME)
 
-        if self._setup_on_boot == SetupOnBootAction.RECONFIG:
-            log.debug("Initial Setup will run in reconfiguration mode.")
-            # write the reconfig trigger file
-            f = open(os.path.join(self._sysroot, "etc/reconfigSys"), "w+")
-            f.close()
+    def _disable_service(self):
+        """Disable the Initial Setup service."""
+        if util.unit_file_exists(self.INITIAL_SETUP_UNIT_NAME, root=self._sysroot):
+            util.disable_service(self.INITIAL_SETUP_UNIT_NAME, root=self._sysroot)
 
-        log.debug("Initial Setup will be enabled.")
-        util.enable_service(unit_name)
+    def _enable_reconfig_mode(self):
+        """Write the reconfig mode trigger file."""
+        log.debug("Initial Setup reconfiguration mode will be enabled.")
+        f = open(os.path.join(self._sysroot, "etc/reconfigSys"), "w+")
```

Comment:
Please, use `with`.

Reply:
Even better, turns out we have a utility function for empty file creation that can be used.